app/Console/Commands/Indexer/IncrementalIndexer.php

```diff
@@ -30,6 +30,23 @@ protected function promptForMissingArgumentsUsing(): array
         ];
     }
 
+    private function sleep(int $milliseconds): void
+    {
+        $interval = 100; // check every 100 ms
+        $elapsed = 0;
+
+        while ($elapsed < $milliseconds)
+        {
+            if ($this->cancelled)
```

Comment:
Should we `return 1` and use `$log->fail()` instead?
https://laravel.com/docs/11.x/artisan#exit-codes

Reply:
Both of us are wrong. https://tldp.org/LDP/abs/html/exitcodes.html
This should be `128 + {the signal received}`

Reply:
The goal is to let the parent process know which signal terminated the process.
E.g. in PHPStorm when I stop the run:
![image](https://github.com/user-attachments/assets/222cb679-f9f6-45d4-be29-c398531e7666)